client/src/main/java/io/pravega/client/stream/StreamCut.java

```diff
@@ -58,6 +58,11 @@ public String toString() {
         private Object readResolve() {
             return UNBOUNDED;
         }
+
+        @Override
+        public int compareTo(StreamCut o) {
```

Comment:
This is `StreamCut.UNBOUNDED`'s compareTo(). UNBOUNDED is used to mean both "possibly ever-truncating stream head" AND "ever-extending stream tail", depending on argument position (whether it is used as a start streamcut or an end streamcut). Given that UNBOUNDED can refer to the head OR the tail, depending on where it is used, how should its compareTo() behave?

Should it be equal only to itself and greater-than all other streamcuts? (That is, should it represent the unbounded tail when compared to?)